third_party/s2n-bignum/include/s2n-bignum_aws-lc.h

```diff
@@ -50,6 +50,19 @@ static inline uint8_t use_s2n_bignum_alt(void) {
 }
 #endif
 
+// Montgomery inverse modulo p_256 = 2^256 - 2^224 + 2^192 + 2^96 - 1
+// Input x[4]; output z[4]
+extern void bignum_montinv_p256(uint64_t z[static 4],const uint64_t x[static 4]);
+
+// Montgomery-Jacobian form scalar multiplication for P-256
+// Input scalar[4], point[12]; output res[12]
+extern void p256_montjscalarmul(uint64_t res[static 12],const uint64_t scalar[static 4],const uint64_t point[static 12]);
+extern void p256_montjscalarmul_alt(uint64_t res[static 12],const uint64_t scalar[static 4],const uint64_t point[static 12]);
+static inline void p256_montjscalarmul_selector(uint64_t res[static 12], const uint64_t scalar[static 4], const uint64_t point[static 12]) {
+  if (use_s2n_bignum_alt()) { p256_montjscalarmul_alt(res, scalar, point); }
+  else { p256_montjscalarmul(res, scalar, point); }
```

Comment:
NP: in a far future change would it make sense to rename these functions something more descriptive like `p256_montjscalarmul_wide_multiplier`? Such that in the future if we have a third implementation it's not `p256_montjscalarmul_alt2` or `p256_montjscalarmul_alt_alt`.